app/javascript/packs/mock-device-profiling.tsx

```diff
@@ -0,0 +1,83 @@
+import { render } from 'react-dom';
+import { useInstanceId } from '@18f/identity-react-hooks';
+import { ChangeEvent } from 'react';
+
+/**
+ * Loads the ?session_id=SESSION_ID param from the <script> tag. Unfortunately
+ * import.meta.url doesn't have live URL params so we need to scrape the DOM.
+ */
+function loadSessionId(): string | undefined {
+  let sessionId;
+  Array.from(document.scripts).every((scriptTag) => {
```

Comment:
Would [`document.currentScript`](https://developer.mozilla.org/en-US/docs/Web/API/Document/currentScript) work here?

Reply:
The docs say that's only for an element that's not for JS modules... and it ended up being `null` for me when I tried it just now

Reply:
If you access it outside of any callbacks it should work, so something like:

```typescript
const { currentScript } = document.currentScript;

function loadSessionId() : string | undefined {
  /* do things with currentScript */
}
```